### PR TITLE
Update reffile docs for msaflagopen and pathloss

### PR DESCRIPTION
**Pull request by hbushouse**
Updated the reference file descriptions for the `msaflagopen` and `pathloss` steps, in fulfillment of #2675. Also includes some minor wording and formatting changes to `mrs_imatch` description, and removes non-existent reference to `NRSFlatModel` in `datamodels`.

### CODE REVIEW

**Comment by hbushouse** [comment]
@stscirij At the end of the `pathloss` reference files description, it looks to me like the descriptions of the WCS axis 1 and 2 keywords for the NIRISS SOSS reffile are cut-and-paste copies of the descriptions from the NIRSpec reffiles, when in fact the SOSS mode is somewhat different (e.g. dimension 2 looks like it's a function of pupil wheel position). Please check those statements and let me know how they should be updated.

File: docs/jwst/pathloss/reference_files.rst [hunk 1]
```diff
@@ -1,137 +1,192 @@
...
-======= ========== =========================================================================================
-
-This dimension expresses the change of correction with wavelength
+CTYPE3  'METER'    Type of physical coordinate in this dimension
+======= ========== ==============================================================================
 
 The NIRISS SOSS reference file should also have WCS components, but their
-interpretation is different from those in the NIRSPEC reference file:
+interpretation is different from those in the NIRSpec reference file.
+Dimension 1 expresses the decenter along the dispersion direction for a point source:
```
Comment (stscirij) [inline docs/jwst/pathloss/reference_files.rst]: This should be:
Dimension 1 expresses the column number in the science data.

File: docs/jwst/pathloss/reference_files.rst [hunk 2]
```diff
@@ -1,137 +1,192 @@
...
 CRPIX1  5.0                   Reference pixel in fastest dimension
 CRVAL1  5.0                   Coordinate value at this reference pixel
 CDELT1  1.0                   Change in coordinate value for unit change in index
 CTYPE1  'PIXEL'               Type of physical coordinate in this dimension
-======= ===================== =========================================================================================
+======= ===================== ===================================================================
 
-This dimension expresses the decenter along the dispersion direction for a point source
+Dimension 2 expresses the decenter along the direction perpendicular to the dispersion
+for a point source:
```
Comment (stscirij) [inline docs/jwst/pathloss/reference_files.rst]: This should be:
Dimension 2 expresses the value of the PWCPOS keyword.

File: docs/jwst/pathloss/reference_files.rst (hunk 1 above)
Comment (hbushouse) [reply docs/jwst/pathloss/reference_files.rst]: Fixed in latest commit

File: docs/jwst/pathloss/reference_files.rst (hunk 2 above)
Comment (hbushouse) [reply docs/jwst/pathloss/reference_files.rst]: Fixed in latest commit.

**Pull request merged**